xbmc/filesystem/FileCache.cpp

```diff
@@ -153,29 +153,40 @@ bool CFileCache::Open(const CURL& url)
     else
     {
       size_t cacheSize;
-      if (m_fileSize > 0 && m_fileSize < CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_cacheMemSize && !(m_flags & READ_AUDIO_VIDEO))
+      if (!(m_flags & READ_AUDIO_VIDEO) && m_fileSize > 0 && m_fileSize < CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_cacheMemSize)
```

Comment:
What about streaming flac audio?

Reply:
And which video do you have in mind < 1 MB (nfs chunk size)? There are possibly more mp3s < 1 MB out there.

Reply:
No we can't since audio/video streams may grow in size.

Reply:
Fine with me. Let's check it out. And get it in.